### PR TITLE
Event modifier cleanup

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

File: src/PublicApiGenerator/EventNameBuilder.cs [hunk 1]
```diff
@@ -39,20 +39,22 @@ public static class EventNameBuilder
...
+                case MemberAttributes.Static when isNew.Value:
                     return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "static new")+ name;
-                case MemberAttributes.Override when addScopeAttributes == MemberAttributes.Override:
+                case MemberAttributes.Override:
                     return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "override")+ name;
-                case MemberAttributes.Final | MemberAttributes.Override when addScopeAttributes == (MemberAttributes.Final | MemberAttributes.Override):
+                case MemberAttributes.Final | MemberAttributes.Override:
                     return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "sealed override")+ name;
-                case MemberAttributes.Final when addScopeAttributes == MemberAttributes.Final && isNew.HasValue && isNew.Value:
+                case MemberAttributes.Final when isNew.HasValue && isNew.Value:
```
Comment (jnm2) [inline src/PublicApiGenerator/EventNameBuilder.cs]: (`isNew.HasValue && isNew.Value` could be `isNew is true`, same other places)

File: src/PublicApiGenerator/EventNameBuilder.cs [hunk 2]
```diff
@@ -39,20 +39,22 @@ public static class EventNameBuilder
             var addScopeAttributes = addAccessorAttributes & MemberAttributes.ScopeMask;
             switch (addScopeAttributes)
             {
-                case MemberAttributes.Static when addScopeAttributes == MemberAttributes.Static && !isNew.HasValue:
+                case MemberAttributes.Static when isNew.HasValue is false:
```
Comment (jnm2) [inline src/PublicApiGenerator/EventNameBuilder.cs]: `isNew.HasValue is false` could be `isNew is null`

**Comment by jnm2** [comment]
@danielmarbach Ugh, I missed `new virtual` too.

File: src/PublicApiGenerator/EventNameBuilder.cs (hunk 2 above)
Comment (danielmarbach) [reply src/PublicApiGenerator/EventNameBuilder.cs]: Dude you are a machine.

**Comment by jnm2** [comment]
I'm not necessarily recommending this, but just so you're aware, this is an option:

```cs
switch ((addAccessorAttributes & MemberAttributes.ScopeMask, isNew))
{
    case (MemberAttributes.Static, null)
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "static") + name;
        
    case (MemberAttributes.Static, true)
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "static new") + name;
        
    case (MemberAttributes.Override, _):
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "override") + name;
        
    case (MemberAttributes.Final | MemberAttributes.Override, _):
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "sealed override") + name;
        
    case (MemberAttributes.Final, true):
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "new") + name;
        
    case (MemberAttributes.Abstract, null):
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "abstract") + name;
        
    case (MemberAttributes.Abstract, true):
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "new abstract") + name;
        
    case (MemberAttributes.Const, _):
        return string.Format(CodeNormalizer.EventModifierMarkerTemplate, "abstract override") + name;
}
```

**Comment by sungam3r** [comment]
Or even this:
```c#
((addAccessorAttributes & MemberAttributes.ScopeMask, isNew)) switch
{
    (MemberAttributes.Static, null) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "static") + name,
        
    (MemberAttributes.Static, true) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "static new") + name,
        
    (MemberAttributes.Override, _) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "override") + name,
        
    (MemberAttributes.Final | MemberAttributes.Override, _) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "sealed override") + name,
        
    (MemberAttributes.Final, true) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "new") + name,
        
    (MemberAttributes.Abstract, null) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "abstract") + name,
        
    (MemberAttributes.Abstract, true) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "new abstract") + name,
        
    (MemberAttributes.Const, _) => string.Format(CodeNormalizer.EventModifierMarkerTemplate, "abstract override") + name,
  (_,_) => throw new Exception("boom")
}
```

**Comment by danielmarbach** [comment]
Ok guys feel free to take this on and extend it to methods and properties. I will take a longer break from this project because I have a few obligations to fulfill.

**Comment by danielmarbach** [comment]
Beta5 is up

**Comment by jnm2** [comment]
> Or even this:

It's not supposed to throw an exception if no case matches; it runs the stuff after the switch.

**Comment by sungam3r** [comment]
I understand, it was just a quick typed example.

**Pull request merged**